### PR TITLE
Update py-requests-cache to 0.5.2

### PR DESCRIPTION
**Pull request by jhoyt4**
#### Description

py-requests-cache: update to 0.5.2

* update to version 0.5.2
* add link to latest package

Closes: https://trac.macports.org/ticket/61448
Closes: https://github.com/macports/macports-ports/pull/9240

###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS 11.0.1 20B29
Xcode 12.2 12B45b

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-requests-cache/Portfile [hunk 1]
```diff
@@ -17,12 +17,14 @@ maintainers         nomaintainer
...
-checksums           rmd160  9d8f8baf17f534d5394acdc12855a892e5ce1578 \
-                    sha256  6822f788c5ee248995c4bfbd725de2002ad710182ba26a666e85b64981866060 \
-                    size    30711
+checksums           rmd160  ccd76aca28b1ba826a88beb5fc65f16aec26e583 \
+                    sha256  813023269686045f8e01e2289cc1e7e9ae5ab22ddd1e2849a9093ab3ab7270eb \
+                    size    31159
 
 python.versions     27 35 36 37 38
 
+master_sites        https://files.pythonhosted.org/packages/0c/d4/bdc22aad6979ceeea2638297f213108aeb5e25c7b103fa02e4acbe43992e
```
Comment (cjones051073) [inline python/py-requests-cache/Portfile:26]: Why is this required ? It shouldn't be necessary, the github PG should handle things like this itself.

Comment (jhoyt4) [reply python/py-requests-cache/Portfile:26]: I couldn't get it to download without the line.  I can try again once my machine frees up from testing qt5.15.2...

Comment (jhoyt4) [reply python/py-requests-cache/Portfile:26]: If you have a suggested edit happy to make it as well - I'm very new to the internals of macports

Comment (cjones051073) [reply python/py-requests-cache/Portfile:26]: If it fails without this line need to see the output to see why. There will likely be better ways to fix it. Please send a log of the output you get from running in verbose mode. ‘sudo port -v .....’

**Comment by cjones051073** [comment]
Your commit message does not follow our guidelines. Please update to it to the form

Portname: description

File: python/py-requests-cache/Portfile (hunk 1 above)
Comment (jhoyt4) [reply python/py-requests-cache/Portfile:26]: > % sudo port -v install py38-requests-cache
--->  Computing dependencies for py38-requests-cache.
--->  Fetching archive for py38-requests-cache
--->  py38-requests-cache-0.5.2_0.darwin_20.noarch.tbz2 doesn't seem to exist in /opt/local/var/macports/incoming/verified
--->  Attempting to fetch py38-requests-cache-0.5.2_0.darwin_20.noarch.tbz2 from https://packages.macports.org/py38-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch py38-requests-cache-0.5.2_0.darwin_20.noarch.tbz2 from https://ywg.ca.packages.macports.org/mirror/macports/packages/py-requests-cache/py38-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch py38-requests-cache-0.5.2_0.darwin_20.noarch.tbz2 from https://mse.uk.packages.macports.org/py38-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Fetching distfiles for py38-requests-cache
--->  requests-cache-0.5.2.tar.gz does not exist in /opt/local/var/macports/distfiles/py-requests-cache
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://github.com/reclosedev/requests-cache/tarball/v0.5.2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   168  100   168    0     0    142      0  0:00:01  0:00:01 --:--:--   142
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://ykf.ca.distfiles.macports.org/MacPorts/mpdistfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://ywg.ca.distfiles.macports.org/mirror/macports/distfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://mse.uk.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://lil.fr.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://cph.dk.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://nue.de.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://fco.it.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://kmq.jp.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://cjj.kr.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://jnb.za.distfiles.macports.org/distfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://aarnet.au.distfiles.macports.org/pub/macports/distfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from https://pek.cn.distfiles.macports.org/macports/distfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://atl.us.distfiles.macports.org/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch requests-cache-0.5.2.tar.gz from http://jog.id.distfiles.macports.org/macports/distfiles/py-requests-cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:30 --:--:--     0
Error: Failed to fetch py38-requests-cache: Connection timed out after 30000 milliseconds
Error: See /opt/local/var/macports/logs/_opt_mports_macports-ports_python_py-requests-cache/py38-requests-cache/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port py38-requests-cache failed

Comment (cjones051073) [reply python/py-requests-cache/Portfile:26]: Ok. So the core issue is ulstream has not tagged these release at

https://github.com/reclosedev/requests-cache

Fetching the tarball from pyhosted is not the best option here. I suggest fetching a specific git hash instead and use that as 0.5.2

Comment (cjones051073) [reply python/py-requests-cache/Portfile:26]: Something like

github.setup        reclosedev requests-cache 6717a336430eb7e6fe17e75c5517f3076ca85b42
version 0.5.2

**Comment by cjones051073** [comment]
I notice you have used the master branch in your fork for this PR. This is a bad idea, you should always create a specific branch for each PR you intend to submit. See https://guide.macports.org/chunked/project.github.html for more details.

File: python/py-requests-cache/Portfile (hunk 1 above)
Comment (jhoyt4) [reply python/py-requests-cache/Portfile:26]: ok - I got that working.  Now need to fight git to rollback the previous commit add the new one and correct the commit message

**Comment by jhoyt4** [comment]
> I notice you have used the master branch in your fork for this PR. This is a bad idea, you should always create a specific branch for each PR you intend to submit. See https://guide.macports.org/chunked/project.github.html for more details.

Thanks - will do moving forward.  I am about two weeks down this rabbit hole and am learning as I go...

File: python/py-requests-cache/Portfile (hunk 1 above)
Comment (cjones051073) [reply python/py-requests-cache/Portfile:26]: git commit --amend

Then

git push --force ....

Comment (jhoyt4) [reply python/py-requests-cache/Portfile:26]: just pushed - I think I got it.

Thank you very much for the assistance!

Comment (cjones051073) [reply python/py-requests-cache/Portfile:26]: Not quite. Check the diffs, it still sets the master site to pyhosted.

Comment (jhoyt4) [reply python/py-requests-cache/Portfile]: sorry about that - I think I got it this time (hopefully...).

Thank you for your patience.

Comment (jhoyt4) [reply python/py-requests-cache/Portfile]: crud - now it deleted my previous edits.  Sorry will fix - this is frustrating.

Comment (jhoyt4) [reply python/py-requests-cache/Portfile]: ok - that got it.  Just needs to get through the last test (CI/build)

File: python/py-requests-cache/Portfile [hunk 2]
```diff
@@ -4,7 +4,8 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           python 1.0
 
-github.setup        reclosedev requests-cache 0.5.0 v
+github.setup reclosedev requests-cache 6717a336430eb7e6fe17e75c5517f3076ca85b42
+version 0.5.2
```
Comment (cjones051073) [inline python/py-requests-cache/Portfile:8]: Please indent these lines to match the rest of the lines in the portfile.

**Review by cjones051073 — CHANGES_REQUESTED** [review]

File: python/py-requests-cache/Portfile (hunk 2 above)
Comment (jhoyt4) [reply python/py-requests-cache/Portfile]: done

**Comment by cjones051073** [comment]
Thanks, looks ok now.

**Review by cjones051073 — APPROVED** [review]
